foundry/clientSettings.d.ts

```diff
@@ -195,7 +199,15 @@ declare namespace ClientSettings {
       step: number;
     };
     scope: string;
-    type?: ConstructorOf<T>;
```

Comment:
Do arrays need to be handled specially here? Also, I don't think `ConstructorOf` is actually the right thing here, as the only allowed values are `Boolean`, `Object`, etc., right? so I would just make it
```ts

    type?: T extends boolean
      ? typeof Boolean
      : T extends number
      ? typeof Number
      : T extends string
      ? typeof String
      : T extends object
      ? typeof Object;
```
Also what about other types (e.g. `symbol`), are they allowed? if not, that should be prevented explicitly.

Reply:
I've only seen those types used for setting in modules. I'll double check what the valid types are. I suppose that the final case could be `never` instead of `ConstructorOf<T>`

Reply:
Looking at foundry.js, the get method calls the constructor defined here on the stored data before returning it (if it's defined). So theoretically any constructor is valid. I'll add `bigint` and `symbol` cases for completeness.